test/integration/fixtures/regression/issue-2286.js

```diff
@@ -0,0 +1,5 @@
+describe('suite', function () {
+  beforeEach(function () { this.skip() })
+  after(function () { console.log('after in suite') })
+  it('test', function () {})
+})
```

Comment:
(Went ahead and committed both; feel free to rebase away either you don't want.)